app/Models/Airline.php

```diff
@@ -11,11 +11,36 @@ class Airline extends Model
 {
     protected $guarded = ['id'];
 
-    public function flights(): HasMany {
+    public function flights(): HasMany
+    {
         return $this->hasMany(Flight::class);
     }
 
-    public function cities(): BelongsToMany {
+    public function cities(): BelongsToMany
+    {
         return $this->belongsToMany(City::class);
     }
+
+    public function scopeOrder(Builder $query, string $sortBy, bool $ascending): void
+    {
+        $order = $ascending ? 'asc' : 'desc';
+
+        $query->orderBy($sortBy, $order);
+    }
+
+    public function scopeFilterByCity(Builder $query, ?int $cityId): void
+    {
+        $query->when($cityId, function($query, $cityId) {
+            $query->whereHas('cities', function($query) use($cityId) {
+                $query->where('city_id', $cityId);
+            });
+        });
+    }
+
+    public function scopeFilterByFlights(Builder $query, ?int $flights): void
+    {
+        $query->when(isset($flights), function ($query) use ($flights) {
+            $query->having('flights_count', '=', $flights);
+        });
+    }
```

Comment:
ideally you should put only reusable and isolated scopes here. you can't call this scope like `Airline::query()->FilterByFlights()`. it will break, because `flights_count` doesnt exists. That column is being created with `withCount('flights')` so your scope is depending on how is built the query is using it. 

I think you can add the withCount to the scope too:

`$query->withCount('flights')->having('flights_count', '=', $flights);`

Reply:
How should I go about conditionally executing this line (so that I only apply this filter when the query param is passed)? Is there something better than using `when`, as it is used in the scopeFilter function?